File: Master_the_Key_ROS_Concepts/Lec1_Create_a_Cakin_Workspace/catkin_ws/src/my_robot_tutorials/src/my_first_node.cpp
#include <ros/ros.h> // using ros library

int main(int argc, char **argv)
{
	ros::init(argc, argv, "my_first_cpp_node"); //you can have same name node running at a time
	ros::NodeHandle nh;

	//use log function
	ROS_INFO("Node has been started");

	//sleep function for this which will use duration
	//ros::Duration(1.0).sleep();

	//ROS_INFO("Exit");
	//
	ros::Rate rate(10);
	while (ros::ok()){
		ROS_INFO("Hello");
		rate.sleep();
	}
}

//make sure to update CMakefile add this
//add_executable(node_cpp src/my_first_node.cpp)
//target_link_libraries(node_cpp ${catkin_LIBRARIES})

File: Master_the_Key_ROS_Concepts/Lec1_Create_a_Cakin_Workspace/catkin_ws/src/my_robot_tutorials/scripts/my_first_node.py
#!/usr/bin/env python3

import rospy

if __name__ == '__main__':
    #initialize node
    rospy.init_node('my_first_python_node')

    #log info
    rospy.loginfo("This node has been started")


    #we have rate of 10 hz
    rate = rospy.Rate(10)

    #It will excute bottom loop with .10s
    while not rospy.is_shutdown():
        rospy.loginfo("Hello")
        rate.sleep()
